Log full back trace of error in unit tests

GraphQL catches exceptions and shows very limited amount of information from the error. This makes debugging unit tests almost impossible. It would be much better to log full back traces from errors.

For example following error message does not help much to understand the actual problem:

```
___________________________________________________ test_uuid ____________________________________________________

    def test_uuid():
        @strawberry.type
        class Query:
            @strawberry.field
            def example_uuid_out(self) -> uuid.UUID:
                return uuid.NAMESPACE_DNS

        schema = strawberry.Schema(Query)

        result = schema.execute_sync("{ exampleUuidOut }")

>       assert not result.errors
E       AssertionError: assert not [GraphQLError('', locations=[SourceLocation(line=1, column=3)], path=['exampleUuidOut'])]
E        +  where [GraphQLError('', locations=[SourceLocation(line=1, column=3)], path=['exampleUuidOut'])] = ExecutionResult(data=None, errors=[GraphQLError('', locations=[SourceLocation(line=1, column=3)], path=['exampleUuidOut'])], extensions={}).errors

tests/schema/types/test_uuid.py:17: AssertionError
```

The error message above was cause by this
```diff
--- a/strawberry/field.py
+++ b/strawberry/field.py
@@ -244,6 +244,7 @@ class StrawberryField(dataclasses.Field):
             return result

         def _strawberry_info_from_graphql(info: GraphQLResolveInfo) -> Info:
+            raise Exception()
             return Info(
                 field_name=info.field_name,
                 field_nodes=info.field_nodes,
```

--- a/RELEASE.md
+++ b/RELEASE.md
@@ -1,0 +1,3 @@
+Release type: minor
+
+Add a `process_errors` method to `strawberry.Schema` which logs all exceptions during execution to a `strawberry.execution` logger.

--- a/strawberry/schema/schema.py
+++ b/strawberry/schema/schema.py
@@ -1,3 +1,4 @@
+import logging
 from typing import Any, Dict, List, Optional, Sequence, Type, Union
 
 from graphql import (
@@ -7,6 +8,7 @@ from graphql import (
     parse,
     validate_schema,
 )
+from graphql.error import GraphQLError
 from graphql.subscription import subscribe
 from graphql.type.directives import specified_directives
 
@@ -21,6 +23,9 @@ from strawberry.union import StrawberryUnion
 from ..middleware import DirectivesMiddleware, Middleware
 from ..printer import print_schema
 from .execute import execute, execute_sync
+
+
+logger = logging.getLogger("strawberry.execution")
 
 
 class Schema:
@@ -83,6 +88,11 @@ class Schema:
 
         return None
 
+    def process_errors(self, errors: List[GraphQLError]) -> None:
+        for error in errors:
+            actual_error = error.original_error or error
+            logger.error(actual_error, exc_info=actual_error)
+
     async def execute(
         self,
         query: str,
@@ -104,6 +114,9 @@ class Schema:
             execution_context_class=self.execution_context_class,
             validate_queries=validate_queries,
         )
+
+        if result.errors:
+            self.process_errors(result.errors)
 
         return ExecutionResult(
             data=result.data,
@@ -132,6 +145,9 @@ class Schema:
             execution_context_class=self.execution_context_class,
             validate_queries=validate_queries,
         )
+
+        if result.errors:
+            self.process_errors(result.errors)
 
         return ExecutionResult(
             data=result.data,

--- a/tests/schema/test_execution.py
+++ b/tests/schema/test_execution.py
@@ -180,7 +180,6 @@ async def test_logging_exceptions(caplog):
     result = await schema.execute(
         query,
         root_value=Query(),
-        validate_queries=False,
     )
 
     assert len(result.errors) == 1
@@ -193,3 +192,35 @@ async def test_logging_exceptions(caplog):
     assert record.message == "test"
     assert record.name == "strawberry.execution"
     assert record.exc_info[0] is ValueError
+
+
+@pytest.mark.asyncio
+async def test_logging_graphql_exceptions(caplog):
+    @strawberry.type
+    class Query:
+        @strawberry.field
+        def example(self) -> int:
+            return None  # type: ignore
+
+    schema = strawberry.Schema(query=Query)
+
+    query = """
+        query {
+            example
+        }
+    """
+
+    result = await schema.execute(
+        query,
+        root_value=Query(),
+    )
+
+    assert len(result.errors) == 1
+
+    # Exception was logged
+    assert len(caplog.records) == 1
+    record = caplog.records[0]
+
+    assert record.levelname == "ERROR"
+    assert record.name == "strawberry.execution"
+    assert record.exc_info[0] is TypeError

--- a/tests/schema/test_execution.py
+++ b/tests/schema/test_execution.py
@@ -159,3 +159,37 @@ async def test_sending_wrong_variables():
             """
         ).strip()
     )
+
+
+@pytest.mark.asyncio
+async def test_logging_exceptions(caplog):
+    @strawberry.type
+    class Query:
+        @strawberry.field
+        def example(self) -> int:
+            raise ValueError("test")
+
+    schema = strawberry.Schema(query=Query)
+
+    query = """
+        query {
+            example
+        }
+    """
+
+    result = await schema.execute(
+        query,
+        root_value=Query(),
+        validate_queries=False,
+    )
+
+    assert len(result.errors) == 1
+
+    # Exception was logged
+    assert len(caplog.records) == 1
+    record = caplog.records[0]
+
+    assert record.levelname == "ERROR"
+    assert record.message == "test"
+    assert record.name == "strawberry.execution"
+    assert record.exc_info[0] is ValueError

